tokio/src/sync/notify.rs

```diff
@@ -101,7 +101,7 @@ use std::task::{Context, Poll, Waker};
 #[derive(Debug)]
 pub struct Notify {
     state: AtomicU8,
-    waiters: Mutex<LinkedList<Waiter>>,
+    waiters: Mutex<LinkedList<Waiter, <Waiter as linked_list::Link>::Target>>,
```

Comment:
Nit, take it or leave it: since this also shows up in the signature of `notify_locked`, I might consider something like 
```rust
type WaitList = LinkedList<Waiter, <Waiter as linked_list::Link>::Target>;

pub struct Notify {
    waiters: Mutex<WaitList>,
    // ...
}

// ...

fn notify_locked(&mut WaitList, /* ... */) -> ...
```

Not a blocker, though